### PR TITLE
Adding headers to projects and applied clang-tidy

### PR DESCRIPTION
**Pull request by bernhardmgruber**
* added a custom target for mallocMC headers, this makes the headers show up in IDE projects (e.g. Visual Studio)
* added header files to their respective projects (exampels and VerifyHeap)
* applied clang-tidy
* using global includes in examples and tests for mallocMC headers

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,9 +71,13 @@ INSTALL(
...
 # Executables
 ###############################################################################
-add_custom_target(examples DEPENDS mallocMC_Example01 mallocMC_Example02 mallocMC_Example03 VerifyHeap)
+file(GLOB_RECURSE headers src/include/**)
+add_custom_target(mallocMC SOURCES ${headers}) # create a target with the header files for IDE projects
+source_group(TREE ${CMAKE_CURRENT_LIST_DIR}/src/include FILES ${headers})
 
-add_executable(mallocMC_Example01 EXCLUDE_FROM_ALL examples/mallocMC_example01.cu)
+include_directories(${CMAKE_CURRENT_LIST_DIR}/src/include)
+add_executable(mallocMC_Example01 EXCLUDE_FROM_ALL examples/mallocMC_example01.cu examples/mallocMC_example01_config.hpp)
```
Comment (bernhardmgruber) [inline CMakeLists.txt:79]: btw, why do you specify the EXCLUDE_FROM_ALL? I know it's a header-only library. So the most users don't need to build anything. So we can just as well build the examples and tests if someone builds the project (either running `make` or hitting build all in an IDE).

Comment (psychocoderHPC) [reply CMakeLists.txt:79]: The CMake is very old. This is the workaround taht the test will not be build when we use mallocMC in a third party project. 
In the past we thought there is no need to install a header only library. This thinking is wrong. We need to refactore the CMake file to allow installing the header only library via  `make install` or the cmake equivalent.

Comment (bernhardmgruber) [reply CMakeLists.txt:79]: I see. Thank you! I think we should properly rework the CMakeLists at some point then. But probably not as part of this PR.

File: src/include/mallocMC/creationPolicies/Scatter_impl.hpp [hunk 2]
```diff
@@ -75,7 +75,7 @@ namespace ScatterKernelDetail{
   {
     public:
       typedef T_Config  HeapProperties;
```
Comment (psychocoderHPC) [inline src/include/mallocMC/creationPolicies/Scatter_impl.hpp:77]: Somehow clang-tidy missed to change this. Could you maybe grep for `typedef` and adjust this and other lines by hand.

Comment (bernhardmgruber) [reply src/include/mallocMC/creationPolicies/Scatter_impl.hpp]: thx, done!

**Pull request merged**